### PR TITLE
SAK-25447 - Update short URLs during site duplication

### PR DESCRIPTION
**Pull request by mpellicer**

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/LinkMigrationHelperImpl.java [hunk 1]
```diff
@@ -150,4 +160,63 @@ private String findLinkContent(String link) throws Exception {
...
+		}
+		return replacedBody;
+	}
+
+	private String expandShortenedUrl(String shortenedUrl){
+		String expandedURL = StringUtils.EMPTY;
+		try{
+			URL url = new URL(shortenedUrl);
+			// open connection
+			HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection(Proxy.NO_PROXY);
```
Comment (ottenhoff) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/LinkMigrationHelperImpl.java]: What if a proxy is being used?

Comment (mpellicer) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/LinkMigrationHelperImpl.java]: You're right @ottenhoff , thanks for the reminder!

Removed the NO_PROXY configuration. Do we need to implement something if a proxy is defined in the JVM properties? I don't think we need but just asking...

**Comment by mpellicer** [comment]
I need to perform tests with bit.ly as provider.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/LinkMigrationHelperImpl.java [hunk 2]
```diff
@@ -150,4 +159,63 @@ private String findLinkContent(String link) throws Exception {
...
+			}
+
+			// js, css, ...
+			for (Element link : imports) {
+				references.add(link.attr("abs:href"));
+			}
+
+			for(String reference : references){
+				//If doesn't contain the prefix /x/, should ignore the URL.
+				if(!reference.contains("/x/")){
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/util/impl/LinkMigrationHelperImpl.java]: Added

**Comment by mpellicer** [comment]
Hey @ern, I refined the check and now grabs bit.ly URLs too, I tested the PR mixing Sakai and bit.ly URLs and works as expected.

**Pull request merged**